core/server/api/themes.js

```diff
@@ -0,0 +1,172 @@
+// # Themes API
+// RESTful API for Themes
+var Promise = require('bluebird'),
+    _ = require('lodash'),
+    gscan = require('gscan'),
+    fs = require('fs-extra'),
+    config = require('../config'),
+    errors = require('../errors'),
+    storage = require('../storage'),
+    settings = require('./settings'),
+    utils = require('./utils'),
+    i18n = require('../i18n'),
+    themes;
+
+/**
+ * ## Themes API Methods
+ *
+ * **See:** [API Methods](index.js.html#api%20methods)
+ */
+themes = {
+    upload: function upload(options) {
+        var zip = {
+            path: options.path,
+            name: options.originalname,
+            shortName: options.originalname.split('.zip')[0]
+        }, theme, storageAdapter = storage.getStorage('themes');
+
+        // check if zip name is casper.zip
+        if (zip.name === 'casper.zip') {
```

Comment:
If we upload CASPER.zip, it would override the current `casper` folder.
I will push a case handling for all uploads.